### ISSUE TITLE
Rename dimension Time to Duration

### ISSUE THREAD

**Issue by rben01** [sharkdp/numbat#631]
I generally think of “time” as meaning time of day — the part of a DateTime that's not the date. Numbat’s usage of it to mean a span of time, i.e., a Duration, is a bit confusing and unexpected.

Were it up to me, I would rename it to Duration.

**Comment by Goju-Ryu** [comment]
I’m a bit conflicted about this. On one hand I agree that having `Time` be the time portion of `DateTime` is reasonable and intuitive. On the other, many physics equations uses a time variable not as duration but as a concrete value on a timeline with time = 0 being the initial state to calculate the evolution of a system over time from. 
I feel like Duration would be counterintuitive here. I think we could use `TimeOfDay`, `ClockTime` or other descriptive names for the time part of `DateTime` to make it obvious which is which. 
I’m not sure I’m decided on my preference but I think I’m leaning toward keeping the current name for time.

**Comment by rben01** [comment]
> On the other, many physics equations uses a time variable not as duration but as a concrete value on a timeline with time = 0 being the initial state to calculate the evolution of a system over time from.

In this case it's still a duration that matters, it's just that the distinction matters less because the duration is formed by subtracting 0 from the instant in question.

**Comment by sharkdp** [comment]
> I’m a bit conflicted about this. On one hand I agree that having `Time` be the time portion of `DateTime` is reasonable and intuitive. On the other, many physics equations uses a time variable not as duration but as a concrete value on a timeline with time = 0 being the initial state to calculate the evolution of a system over time from.

Yes, the name is inspired by physics. Where `Length`, `Mass` and `Time` are the fundamental kinematic dimensions. I'd rather not change the name to `Time`, but we can consider adding `Duration` as an alias.

On the other hand, we might want to properly support durations in `DateTime` calculations. And then it might not be enough to only think of a duration as a single value (in seconds). See e.g. how the datetime library that we use, `jiff`, represents durations: https://docs.rs/jiff/latest/jiff/struct.Span.html
